tests/bpftrace.cpp

```diff
@@ -1909,4 +1911,157 @@ TEST(bpftrace, print_hist_map)
+    std::string name;
+    uint32_t top;
+    uint32_t div;
+    std::string expected_output;
+  };
+
+  constexpr uint64_t base_time_ns = 173482610888837;
+  constexpr int ns_in_ms = 1'000'000;
+  constexpr int interval = ns_in_ms;
+  constexpr uint64_t first_epoch = base_time_ns / interval;
+
+  std::vector<std::pair<EpochType, ValueType>> v1 = {
+    { (first_epoch + 1), { 1, 0, 0, 0, 0, 0, 0, 0 } },
+    { (first_epoch + 2), { 2, 0, 0, 0, 0, 0, 0, 0 } },
+    { (first_epoch + 3), { 3, 0, 0, 0, 0, 0, 0, 0 } },
+    { (first_epoch + 4), { 4, 0, 0, 0, 0, 0, 0, 0 } },
+    { (first_epoch + 5), { 5, 0, 0, 0, 0, 0, 0, 0 } },
+  };
+
+  std::vector<std::pair<EpochType, ValueType>> v2 = {
+    { (first_epoch + 2), { 1, 0, 0, 0, 0, 0, 0, 0 } },
+    { (first_epoch + 3), { 2, 0, 0, 0, 0, 0, 0, 0 } },
+    { (first_epoch + 4), { 3, 0, 0, 0, 0, 0, 0, 0 } },
+    { (first_epoch + 5), { 4, 0, 0, 0, 0, 0, 0, 0 } },
+    { (first_epoch + 6), { 5, 0, 0, 0, 0, 0, 0, 0 } },
+  };
+
+  TSeriesMap values_by_key;
+  values_by_key[{ 0, 0, 0, 0, 0, 0, 0, 0 }] = TSeries(v1.begin(), v1.end());
+  values_by_key[{ 1, 0, 0, 0, 0, 0, 0, 0 }] = TSeries(v2.begin(), v2.end());
+
+  auto map_info = MapInfo{ .key_type = CreateInt64(),
+                           .value_type = CreateTSeries(),
+                           .detail = TSeriesArgs{ .interval_ns = ns_in_ms,
+                                                  .buckets = 5,
+                                                  .agg = CreateInt64() },
+                           .id = {},
+                           .is_scalar = true };
+
+  std::vector<TestCase> test_cases = {
+    // Test case 1: print all buckets
+    { .name = "tseries_map_1",
+      .top = 0,
+      .div = 0,
+      .expected_output = R"(tseries_map_1:
+hh:mm:ss.mmm  ___________________________________________________
+%H:%M:49.611 o                                                   | 2
+%H:%M:49.612 |                o                                  | 3
+%H:%M:49.613 |                                 o                 | 4
+%H:%M:49.614 |                                                   o 5
+%H:%M:49.615 |                                                   | -
+             v___________________________________________________v
+             2                                                   5
+
+tseries_map_1:
+hh:mm:ss.mmm  ___________________________________________________
+%H:%M:49.611 o                                                   | 1
+%H:%M:49.612 |            o                                      | 2
+%H:%M:49.613 |                         o                         | 3
+%H:%M:49.614 |                                      o            | 4
+%H:%M:49.615 |                                                   o 5
+             v___________________________________________________v
+             1                                                   5
+
+)" },
+    // Test case 2: print top 1 bucket
+    { .name = "tseries_map_2",
+      .top = 1,
```

Comment:
What's the difference between this and the one above with the top set to `1`. Does a top arg even make sense for the time series? `div` is also kinda questionable...

Reply:
I was trying to come up with a use for `top` for `tseries`, since it is already passed through by print and ended up sorting keys in the map by "most recently updated tseries". This seemed semi-useful, as you'd generally not care about time series where nothing is happening and might want to truncate the list when printing. 

As for `div`, I would be fine just not supporting this for tseries. I'm not sure it adds much.

Reply:
Removed support for `div` and `top` for now as discussed.